tests/MongoDB.Driver.Tests/Specifications/Runner/BaseTestRunner.cs

```diff
@@ -0,0 +1,391 @@
+﻿/* Copyright 2019-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using FluentAssertions;
+using MongoDB.Bson;
+using MongoDB.Bson.TestHelpers.JsonDrivenTests;
+using MongoDB.Driver.Core;
+using MongoDB.Driver.Core.Bindings;
+using MongoDB.Driver.Core.Clusters.ServerSelectors;
+using MongoDB.Driver.Core.Events;
+using MongoDB.Driver.Core.Misc;
+using MongoDB.Driver.Core.TestHelpers;
+using MongoDB.Driver.Core.TestHelpers.JsonDrivenTests;
+using MongoDB.Driver.Core.TestHelpers.XunitExtensions;
+using MongoDB.Driver.TestHelpers;
+using MongoDB.Driver.Tests.JsonDrivenTests;
+using Xunit;
+
+namespace MongoDB.Driver.Tests.Specifications.Runner
+{
+    public abstract class BaseTestRunner
+    {
+        // constants
+        private string _collectionNameKey = "collection_name";
+        private string _databaseNameKey = "database_name";
+        private readonly string _dataKey = "data";
+        private readonly string _expectationsKey = "expectations";
+        private readonly string _failPortKey = "failPoint";
```

Comment:
Here `Key` means the same as `name`. I took in my mind this construction: "key" : "value". But maybe `name` sounds better.
I implemented it in this way since I wanted to have the ability to overwrite json keys in case if it will be necessary. 
Should I rename `Key` to `Name`, .. or delete this overwrite ability at all?